### PR TITLE
deprecated erlang get stacktrace function

### PR DESCRIPTION
**Pull request by getong**

### CODE REVIEW

File: test/sbroker_SUITE.erl [hunk 1]
```diff
@@ -204,7 +204,12 @@ skip_down_match(_) ->
             Self = self(),
             {await, Ref, _} = sbroker:async_ask_r(Broker, Self, {Self, Ref}),
             sys:resume(Broker),
-            {drop, _} = sbroker:await(Ref, ?TIMEOUT),
+            case sbroker:await(Ref, ?TIMEOUT) of
+                {drop, _} ->
+                    ok;
+                {go, _Ref, _Value, _RelativeTime, _SojournTime} ->
+                    ok
+            end,
```
Comment (fishcakez) [inline test/sbroker_SUITE.erl]: This test is specifically testing that we get a drop here so if this is required there is a bug before this in the test or a bug in the implementation itself.

**Review by fishcakez — COMMENTED** [review]
Stacktrace changes look good but there is issue with unrelated test change.

File: test/sbroker_SUITE.erl (hunk 1 above)
Comment (getong) [reply test/sbroker_SUITE.erl]: ok, I will check it later.

**Comment by fishcakez** [comment]
One thing, I don't really plan to maintain this project as open source any more. New features and some other improvements have remained closed source due to lack of collaborators making it not worthwhile to do the extra work required to managed this as an open source project.

**Comment by getong** [comment]
I can help to maintain it, I can learn to maintain it.

File: rebar.config [hunk 2]
```diff
@@ -4,10 +4,8 @@
 
 {profiles, [
     {test, [{deps, [
-        {proper, ".*",
-            {git, "git://github.com/manopapad/proper.git", {tag, "v1.2"}}},
-        {cth_readable, ".*",
-            {git, "https://github.com/ferd/cth_readable.git", {ref, "0fc53d1"}}}
+        {proper, "1.3.0"},
+        {cth_readable, "1.4.2"}
```
Comment (ferd) [inline rebar.config]: rebar3 now ships with cth_readable so you likely no longer need to include the lib and add the hook by hand

**Comment by getong** [comment]
The `skip_down_match` test fail on the `monitor` function, I make a prototype of this test, and it fails on erlang 21.

``` erlang
-module(a).

-export([start/0]).

start() ->
    Pid = spawn(fun() -> t() end),
    register(t_name, Pid),
    spawn(fun() -> t1() end),
    timer:sleep(1000).

t1() ->
    {_, _MRef} = spawn_monitor(fun() ->
                                       Pid = whereis(t_name),
                                       Pid ! {pid, self()},
                                       exit(normal)
                               end).
t() ->
    receive
        {pid, Pid} ->
            Ref = monitor(process, Pid),
            case demonitor(Ref, [flush, info]) of
                true ->
                    io:format("result is true");
                false ->
                    io:format("result is false")
            end
    end.
```
and run it like this:
```
$ /usr/local/otp_src_21.1.4/bin/erlc a.erl
$ /usr/local/otp_src_21.1.4/bin/erl -s a start -s init stop -noinput
result is true
$ rm a.beam
$ /usr/local/otp_src_20.3.8.15/bin/erlc a.erl
$ /usr/local/otp_src_20.3.8.15/bin/erl -s a start -s init stop -noinput
result is false 
```
I ask this question on erlang-question mail list, and the code is correct.
I currently have no idea of how to fix it.

**Comment by g-andrade** [comment]
@getong 
> The `skip_down_match` test fail on the `monitor` function, I make a prototype of this test, and it fails on erlang 21.

Signal handling was [changed](https://www.youtube.com/watch?v=U-JIzj07uQA) in OTP 21. I suspect the following race condition is the culprit:
1) broker process gets suspended
2) ask request gets enqueued
3) asker process dies
4) broker process gets resumed
5) broker process receives ask request and creates monitor of asker process
6) behaviour now differs:
  - **Up to OTP 21**: broker process receives monitor message telling it asker process has died ✔️
  - **In OTP 21**: broker process receives bid request and successfully matches it with asker ❌

If we focus on the moment the bid is dispatched:
```erlang
205             {await, Ref, _} = sbroker:async_ask_r(Broker, Self, {Self, Ref}),                                                                                                                                                             
206             sys:resume(Broker), 
```

..and reverse the order of the clauses:
```erlang
205             sys:resume(Broker), 
206             {await, Ref, _} = sbroker:async_ask_r(Broker, Self, {Self, Ref}),                                                                                                                                                             
```

The test case passes, as we're giving the broker a chance to start monitoring the (dead) asker process earlier and therefore receive the monitor message before the bid request.

But it's still nondeterministic, we're only tweaking the odds, so I'm not sure it's the best approach. Ditto for a well placed `timer:sleep/1`.

**Pull request closed without merging**